[bug] Exec Exception thrown when a string lenght limit (32k) is hit for PR text being transposed to Jira

: java.util.concurrent.ExecutionException: RestClientException{statusCode=Optional.of(400), errorCollections=[ErrorCollection{status=400, errors={description=The entered text is too long. It exceeds the allowed limit of 32,767 characters.}, errorMessages=[]}]}
	at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:396)
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2096)
	at io.atlassian.util.concurrent.Promises$OfStage.get(Promises.java:357)
	at com.atlassian.jira.rest.client.internal.async.DelegatingPromise.get(DelegatingPromise.java:106)
	at io.quarkus.githubtojira.JiraService.createJira(JiraService.java:174)
	at io.quarkus.githubtojira.JiraService_ClientProxy.createJira(Unknown Source)
	at io.quarkus.githubtojira.GithubToJiraResource.performImport(GithubToJiraResource.java:99)
	at io.quarkus.githubtojira.GithubToJiraResource_ClientProxy.performImport(Unknown Source)
	at io.quarkus.githubtojira.GithubToJiraResource$quarkusrestinvoker$performImport_b4f9a47e116e35386fe88adadc40b2c9a882aa79.invoke(Unknown Source)
	at org.jboss.resteasy.reactive.server.handlers.InvocationHandler.handle(InvocationHandler.java:29)
	at io.quarkus.resteasy.reactive.server.runtime.QuarkusResteasyReactiveRequestContext.invokeHandler(QuarkusResteasyReactiveRequestContext.java:190)
	at org.jboss.resteasy.reactive.common.core.AbstractResteasyReactiveContext.run(AbstractResteasyReactiveContext.java:147)
	at io.quarkus.vertx.core.runtime.VertxCoreRecorder$15.runWith(VertxCoreRecorder.java:677)
	at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2651)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2630)
	at org.jboss.threads.EnhancedQueueExecutor.runThreadBody(EnhancedQueueExecutor.java:1622)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1589)
	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:11)
	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:11)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: RestClientException{statusCode=Optional.of(400), errorCollections=[ErrorCollection{status=400, errors={description=The entered text is too long. It exceeds the allowed limit of 32,767 characters.}, errorMessages=[]}]}
	at com.atlassian.jira.rest.client.internal.async.AbstractAsynchronousRestClient$2.apply(AbstractAsynchronousRestClient.java:189)
	at com.atlassian.jira.rest.client.internal.async.AbstractAsynchronousRestClient$2.apply(AbstractAsynchronousRestClient.java:183)
	at com.atlassian.httpclient.api.ResponsePromiseMapFunction.apply(ResponsePromiseMapFunction.java:51)
	at com.atlassian.httpclient.api.ResponsePromiseMapFunction.apply(ResponsePromiseMapFunction.java:10)
	at io.atlassian.util.concurrent.Promises$OfStage.lambda$fold$4(Promises.java:332)
	at io.atlassian.util.concurrent.Promises.lambda$biFunction$7(Promises.java:422)
	at java.base/java.util.concurrent.CompletableFuture.uniHandle(CompletableFuture.java:934)
	at java.base/java.util.concurrent.CompletableFuture$UniHandle.tryFire(CompletableFuture.java:911)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
	at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2179)
	at com.atlassian.httpclient.apache.httpcomponents.SettableFuturePromiseHttpPromiseAsyncClient$1.lambda$doCompleted$0(SettableFuturePromiseHttpPromiseAsyncClient.java:44)
	at com.atlassian.httpclient.apache.httpcomponents.SettableFuturePromiseHttpPromiseAsyncClient.runInContext(SettableFuturePromiseHttpPromiseAsyncClient.java:91)
	at com.atlassian.httpclient.apache.httpcomponents.SettableFuturePromiseHttpPromiseAsyncClient$ThreadLocalDelegateRunnable.run(SettableFuturePromiseHttpPromiseAsyncClient.java:161)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	... 1 more


Example: this one - 
https://github.com/quarkusio/quarkus/pull/53584

--- a/src/main/java/io/quarkus/githubtojira/JiraService.java
+++ b/src/main/java/io/quarkus/githubtojira/JiraService.java
@@ -160,6 +160,12 @@ public class JiraService {
             prTitle = "[TESTING, PLEASE IGNORE] " + prTitle;
             description = "IGNORE: I'm just testing a new JIRA import app\n\n " + description;
         }
+        // the maximum allowed length by Jira is 32767... leave some extra reserve
+        if(description.length() >= 32600) {
+            Log.warn("Truncating the description of PR " +  prUrl + " to 32600 characters " +
+                    "because it's too long for Jira (original length: " + description.length() + ")");
+            description = description.substring(0, 32600);
+        }
         IssueInput input = new IssueInputBuilder()
                 .setProjectKey(jiraProject)
                 .setSummary(prTitle)

